pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -1140,6 +1206,28 @@ ASTRead2NextOccurrence Read2NextOccurrence(String target) :
 	{ jjtThis.setImage(sb.toString()) ;  jjtThis.value = sb.toString(); return jjtThis ;}
 }
 
+/*
+ Read Tokens up to but not including the target token.
+*/
+ASTRead2NextTokenOccurrence Read2NextTokenOccurrence(int target) :
```

Comment:
This will create a new node in the AST, called "Read2NextTokenOccurrence". This sound quite unusual.
Let's make this a #void node, that returns a string, e.g.

```java
private String Read2NextTokenOccurrence(int target) #void :
{
    StringBuilder sb = new StringBuilder();
    Token nextToken = getToken(1);
    while (nextToken.kind!=target
        && (null == nextToken.specialToken || nextToken.specialToken.kind!=target ) //In case the target is a Special Token
        && nextToken.kind!=EOF) {
        nextToken = getNextToken();
        sb.append(nextToken.getImage());
        sb.append(' ');
        nextToken = getToken(1);
    }
}
{
	{ return sb.toString(); }
}
```

The string can be directly added to the image of SqlPlusCommand (L416):

```
 { sb.append(token.getImage()) ; sb.append(" ") ; sb.append(read2NextTokenOccurrence(EOL)) ;
```

Then you also don't need to modify PLSQLParserVisitorAdapter and AbstractPLSQLRule.

(Note: The nodes ASTSkip2NextTokenOccurrence and so on should have been implemented the same way - these nodes don't make much sense in the AST, but they exist already. But we should try to avoid more of these if possible).

Reply:
Note that there's the Javacc `JAVACODE` keyword for that. This way you don't need 2 blocks:

```java
JAVACODE
private String Read2NextTokenOccurrence(int target) #void {
    StringBuilder sb = new StringBuilder();
    Token nextToken = getToken(1);
    while (nextToken.kind!=target
        && (null == nextToken.specialToken || nextToken.specialToken.kind!=target ) //In case the target is a Special Token
        && nextToken.kind!=EOF) {
        nextToken = getNextToken();
        sb.append(nextToken.getImage());
        sb.append(' ');
        nextToken = getToken(1);
    }
    return sb.toString();
}
```